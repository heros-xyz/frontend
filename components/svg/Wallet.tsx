import { Icon, IconProps } from "@chakra-ui/react";

export const WalletIcon: React.FC<IconProps> = (props) => {
  return (
    <Icon
      {...props}
      viewBox="0 0 52 52"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle cx="26" cy="26" r="26" fill="#FFC0F0" />
      <path
        d="M14 23.3332H38M19.3333 29.9998H20.6667M26 29.9998H27.3333M18 35.3332H34C36.2091 35.3332 38 33.5423 38 31.3332V20.6665C38 18.4574 36.2091 16.6665 34 16.6665H18C15.7909 16.6665 14 18.4574 14 20.6665V31.3332C14 33.5423 15.7909 35.3332 18 35.3332Z"
        stroke="#1E16C1"
        strokeWidth="2"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </Icon>
  );
};
