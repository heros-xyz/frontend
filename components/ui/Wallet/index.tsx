import {
  Box,
  Button,
  Heading,
  Text,
  Flex,
  SkeletonCircle,
  Skeleton,
  Stack,
} from "@chakra-ui/react";
import React from "react";
import { Else, If, Then } from "react-if";
import { formatMoney } from "@/utils/functions";

interface WalletProps {
  title: string;
  currentMoney: number;
  feePrice: number;
  timeReceive: string;
  havePaymentMethod: boolean;
  buttonContent?: string;
  onWithDrawMoney?: () => void;
  isLoading: boolean;
}

const Wallet: React.FC<WalletProps> = ({
  title,
  currentMoney,
  feePrice,
  buttonContent,
  onWithDrawMoney,
  isLoading,
}) => {
  return (
    <If condition={!isLoading}>
      <Then>
        <Box
          bg="accent.1"
          borderRadius={{ base: "lg", xl: "xl" }}
          px={{ base: 5, xl: 7 }}
          py={{ base: 3, xl: 5 }}
        >
          <Heading
            mb="1.5"
            color="primary"
            fontSize={{ base: "md", xl: "xl" }}
            fontWeight="extrabold"
            textTransform="capitalize"
            lineHeight="140%"
          >
            {title}
          </Heading>
          <Text
            mb="1.5"
            fontSize={{ base: "xs", xl: "md" }}
            fontWeight="normal"
            color="primary"
          >
            You currently have{" "}
            <SkeletonCircle as="b" isLoaded={!isLoading}>
              {formatMoney(currentMoney)}
            </SkeletonCircle>{" "}
            in your wallet.
            {currentMoney != 0 && (
              <>
                <br />({feePrice}% fee deducted)
              </>
            )}
          </Text>
          <Text
            fontSize={{ base: "xs", xl: "md" }}
            fontWeight="normal"
            color="primary"
          >
            {currentMoney != 0 &&
              `This can be transferred to your bank account.`}
          </Text>
          <Flex justify={{ base: "center", xl: "flex-end" }}>
            <Button
              variant="primary"
              width={{ base: "100%", xl: "auto" }}
              fontSize={{ base: "md", xl: "xl" }}
              h="48px"
              mt={{ base: 5, xl: 3.5 }}
              isDisabled={currentMoney == 0}
              onClick={onWithDrawMoney}
            >
              {buttonContent}
            </Button>
          </Flex>
        </Box>
      </Then>
      <Else>
        <Box
          bg="accent.1"
          borderRadius={{ base: "lg", xl: "xl" }}
          px={{ base: 5, xl: 7 }}
          py={{ base: 3, xl: 5 }}
        >
          <Heading
            mb="1.5"
            color="primary"
            fontSize={{ base: "md", xl: "xl" }}
            fontWeight="extrabold"
            textTransform="capitalize"
            lineHeight="140%"
          >
            {title}
          </Heading>
          <Stack>
            <Skeleton height="24px" w="60%" />
          </Stack>
          <Flex justifyContent="end" textAlign="right" mt={3}>
            <Skeleton
              height="48px"
              w={{ base: "100%", lg: "225px" }}
              rounded="8px"
            />
          </Flex>
        </Box>
      </Else>
    </If>
  );
};

export default Wallet;
