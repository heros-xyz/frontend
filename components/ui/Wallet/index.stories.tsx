import type {Meta, StoryObj} from "@storybook/react";
import Wallet from "./index";

const meta: Meta<typeof Wallet> = {
    title: "Athlete/Wallet",
    component: Wallet,
    tags: ["autodocs"],
    parameters: {
        layout: "fullscreen",
    },
        argTypes: {
            buttonContent: {
                if: {arg: "havePaymentMethod", truthy: false}
            },
        },
};

export default meta;
type Story = StoryObj<typeof Wallet>;

export const WalletTemplate: Story = {
    args: {
        title: "Wallet",
        currentMoney: 4.5,
        feePrice: 5,
        timeReceive: "12/12/2022",
        havePaymentMethod: true,
        buttonContent: "Add Payment Method"
    },
};
