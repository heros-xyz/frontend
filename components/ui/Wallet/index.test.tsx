import { render } from "@testing-library/react";
import Wallet from "./index";
it("renders Wallet unchanged", () => {
  const { container } = render(
    <Wallet title="Wallet" currentMoney={6.5} havePaymentMethod={true} feePrice={5} timeReceive={"20/10/2022"}/>
  );
  expect(container).toMatchSnapshot();
});
