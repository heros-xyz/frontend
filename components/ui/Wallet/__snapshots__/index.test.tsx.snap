// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Wallet unchanged 1`] = `
<div>
  <div
    class="css-coa58c"
  >
    <h2
      class="chakra-heading css-1dy9deq"
    >
      Wallet
    </h2>
    <p
      class="chakra-text css-sxcrnt"
    >
      You currently have
       
      <b
        class="chakra-skeleton css-1uksm7w"
      >
        $6.50
      </b>
       
      in your wallet.
      <br />
      (
      5
      % fee deducted)
    </p>
    <p
      class="chakra-text css-1ew9a1o"
    >
      This can be transferred to your bank account.
    </p>
    <div
      class="css-1on1je4"
    >
      <button
        class="chakra-button css-1gsa7j5"
        type="button"
      />
    </div>
  </div>
</div>
`;
